contracts/Registry.vy

```diff
@@ -0,0 +1,211 @@
+# @version 0.2.8
+
+interface DetailedERC20:
+    def name() -> String[52]: view
+    def symbol() -> String[30]: view
+
+
+interface Vault:
+    def token() -> address: view
+    def apiVersion() -> String[28]: view
+    def governance() -> address: view
+    def initialize(
+        token: address,
+        governance: address,
+        name: String[64],
+        symbol: String[32],
+        guardian: address,
+    ): nonpayable
+
+
+# len(Vault.releases)
+nextRelease: public(uint256)
+releases: public(HashMap[uint256, address])
+
+# Token.address => len(Vault.deployments)
+nextDeployment: public(HashMap[address, uint256])
+vaults: public(HashMap[address, HashMap[uint256, address]])
+
+# 2-phase commit
+governance: public(address)
+pending_governance: address
+
+
+event NewRelease:
+    event_id: indexed(uint256)
+    template: address
+    api_version: String[28]
+
+event NewVault:
+    token: indexed(address)
+    event_id: indexed(uint256)
+    vault: address
+    api_version: String[28]
```

Comment:
Add `tags`. A vault can have a string of `tags` assigned which follow some convention. Something like `test`, `live`, `dead` and so on. An enum could be a good alternative, but imo a text tag is more flexible.